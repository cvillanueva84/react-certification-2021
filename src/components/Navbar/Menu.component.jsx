import React, { useState } from 'react';
import MenuItem from '@material-ui/core/MenuItem';
import { withStyles } from '@material-ui/core/styles';
import Menu from '@material-ui/core/Menu';
import Login from '../Login';
import { useAuth } from '../../providers/Auth';

const MenuLogin = ({ anchor, toggleMenu }) => {
  const StyledMenu = withStyles({
    paper: {
      border: '1px solid #d3d4d5',
    },
  })((props) => (
    <Menu
      elevation={0}
      getContentAnchorEl={null}
      anchorOrigin={{
        vertical: 'bottom',
        horizontal: 'center',
      }}
      transformOrigin={{
        vertical: 'top',
        horizontal: 'center',
      }}
      {...props}
    />
  ));

  const [loginForm, setLoginForm] = useState(false);

  const showLogin = () => {
    toggleMenu(null);
    setLoginForm(!loginForm);
  };

  const { authenticated, logout } = useAuth();

  const handleClose = () => {
    toggleMenu(null);
  };

  const handleLogout = () => {
    logout();
    toggleMenu(null);
  };

  return (
    <>
      <Login show={loginForm} toggleShow={showLogin} />
      <StyledMenu
        anchorEl={anchor}
        keepMounted
        open={Boolean(anchor)}
        onClose={handleClose}
      >
        {authenticated ? (
          <MenuItem onClick={handleLogout}>Log Out</MenuItem>
        ) : (
          <MenuItem onClick={showLogin}>Log In</MenuItem>
        )}
      </StyledMenu>
    </>
  );
};

export default MenuLogin;
