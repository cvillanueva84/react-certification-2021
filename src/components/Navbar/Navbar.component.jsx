import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import { NavText } from './Navbar.styles.jsx';
import { useStoreContext } from '../../state/Store.provider';

function Navbar() {
  const { store } = useStoreContext();
  const [sidebar, setSidebar] = useState(false);

  const showSidebar = () => setSidebar(!sidebar);
  return (
    <>
      <Link to="#">
        <img
          id="menu"
          data-testid="menu"
          src="menu_icon.png"
          alt="page not found"
          onClick={showSidebar}
        />
      </Link>
      <nav className={sidebar ? 'nav-menu active' : 'nav-menu'}>
        <ul onClick={showSidebar}>
          <Link to="#">
            <img id="menu" src="close-window-32.png" alt="page not found" />
          </Link>
          {store.sidebarData.map((item, index) => {
            return (
              <NavText key={index}>
                <Link to={item.path}>{item.title}</Link>
              </NavText>
            );
          })}
        </ul>
      </nav>
    </>
  );
}

export default Navbar;
