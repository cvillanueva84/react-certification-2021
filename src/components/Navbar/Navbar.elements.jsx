import React from 'react';
import styled from 'styled-components';
import { AiFillYoutube } from 'react-icons/ai';
import { FaSearch } from 'react-icons/fa';

export const Nav = styled.nav`
  background: ${(props) => props.theme.background_color};
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: 100vw;
  height: ${(props) => props.theme.navbar_height};
  position: fixed;
  top: 0;
  z-index: 99;
  padding: 0rem 1.5rem;
`;

const LogoContainer = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: flex-start;
  align-items: center;
  user-select: none;
`;

const LogoText = styled.h1`
  color: ${(props) => props.theme.text_color};
  margin-left: 0.5rem;
  font-size: 2rem;
  font-weight: 900;
  @media (max-width: 600px) {
    display: none;
  }
`;

export function IconLogo() {
  return (
    <LogoContainer>
      <AiFillYoutube size="3.5rem" color="red" />
      <LogoText>WizeTube</LogoText>
    </LogoContainer>
  );
}

const FormSearch = styled.form`
  display: flex;
  display: row;
`;

const InputSearch = styled.input`
  padding: 0rem 0.5rem;
  border: 1px solid ${(props) => props.theme.icon_color};
  margin-left: 1rem;
  width: 350px;

  @media (max-width: 737px) {
    width: 200px;
  }
`;

const SearchButton = styled.button`
  color: ${(props) => props.theme.icon_color};
  display: flex;
  justify-content: center;
  align-content: center;
  padding: 0.5rem 0.75rem;
  margin-right: 1rem;
`;

export function SearchBar() {
  return (
    <FormSearch>
      <InputSearch type="text" placeholder="Search" />
      <SearchButton>
        <FaSearch size="1rem" />
      </SearchButton>
    </FormSearch>
  );
}

export const ProfileImg = styled.img`
  border-radius: 50%;
  width: 3.5rem;
  height: auto;
`;
