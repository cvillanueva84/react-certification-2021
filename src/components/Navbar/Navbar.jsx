import React, { useState } from 'react';
import { GiHamburgerMenu } from 'react-icons/gi';
import { BiSearchAlt2 } from 'react-icons/bi';
import {
  Nav,
  Menu,
  Input,
  ContainerInput,
  InputCheck,
  Image,
  Menu2,
  ButtonMenu,
  NavMenu,
  NavItem,
  DivItem,
  SpanItem,
} from './Navbar.styled';

const Navbar = () => {
  const [click, setClick] = useState(false);
  const handleClick = () => setClick(!click);

  return (
    <>
      <Nav>
        <Menu>
          <NavMenu onClick={handleClick} click={click}>
            <NavItem>
              <DivItem>
                <SpanItem>Home</SpanItem>
              </DivItem>
            </NavItem>
          </NavMenu>
          <ButtonMenu onClick={handleClick}>
            <GiHamburgerMenu />
          </ButtonMenu>
          <ContainerInput>
            <BiSearchAlt2 style={{ marginRight: '8px' }} />
            <Input placeholder="Search..." />
          </ContainerInput>
        </Menu>
        <Menu2>
          <InputCheck type="checkbox" />
          <span>Dark mode</span>
          <Image>
            <img
              src="https://w7.pngwing.com/pngs/340/956/png-transparent-profile-user-icon-computer-icons-user-profile-head-ico-miscellaneous-black-desktop-wallpaper.png"
              alt="Fotografia"
            />
          </Image>
        </Menu2>
      </Nav>
    </>
  );
};
export default Navbar;
