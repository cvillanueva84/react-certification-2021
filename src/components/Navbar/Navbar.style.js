import { makeStyles } from '@material-ui/core/styles';

export const useStyles = makeStyles((theme) => ({
    grow: {
        flexGrow: 1,
    },
    title: {
        display: 'none',
        [theme.breakpoints.up('sm')]: {
          display: 'block',
        },
    },
    sectionDesktop: {
        display: 'none',
        [theme.breakpoints.up('md')]: {
          display: 'flex',
        },
    },
    sectionMobile: {
        display: 'flex',
        [theme.breakpoints.up('md')]: {
            display: 'none',
        },
    }
}));
