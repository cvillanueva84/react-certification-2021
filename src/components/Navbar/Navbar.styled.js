import styled from 'styled-components';

export const Nav = styled.div`
  width: 100%;
  height: 64px;
  background-color: rgb(33, 85, 117);
  margin: 0;
  padding: 0;
  display: flex;
  align-items: center;
  justify-content: space-between;
  box-shadow: 0 5px 10px rgba(0, 0, 0, 0.19), 0 5px 5px rgba(0, 0, 0, 0.23);
`;

export const Menu = styled.div`
  /* background-color: red; */
  height: 50px;
  width: 25%;
  display: flex;
  align-items: center;
  justify-content: flex-start;
  color: white;
  font-size: 20px;
  font-weight: lighter;
  font-family: 'Roboto', sans-serif;
`;
export const Menu2 = styled.div`
  margin-right: 22px;
  /* background-color: blue; */
  height: 50px;
  width: 20%;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  color: white;
  font-size: 16px;
  font-weight: lighter;
  font-family: 'Roboto', sans-serif;
  @media screen and (max-width: 960px) {
    display: none;
  }
`;
export const ContainerInput = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  border-radius: 4px;
  background-color: rgba(255, 255, 255, 0.15);
  height: 35px;
  color: white;
  padding-left: 20px;

  font-size: 22px;
`;

export const Input = styled.input`
  border: none;
  background: transparent;
  outline: none;

  width: 180px;

  &::placeholder {
    color: rgb(142, 169, 185);
    font-size: 16px;
    font-weight: lighter;
    font-family: 'Roboto', sans-serif;
  }
`;

export const InputCheck = styled.input`
  margin: 10px;
  display: flex;
  align-items: center;
  appearance: none;
  width: 40px;
  height: 13px;
  border-radius: 20px;
  background: rgb(21, 54, 73);
  position: relative;
  box-sizing: border-box;

  &::before {
    content: '';
    width: 20px;
    height: 20px;
    background: #ffffff;
    border-radius: 50%;
    position: absolute;
    transition: all 0.3s ease-in-out;
  }

  &:checked {
    background: rgb(60, 120, 169);
    &::before {
      background: rgb(37, 133, 123);
      transform: translate(100%, 5%);
      transition: all 0.3s ease-in-out;
      box-shadow: 0px 1px 0px 1px rgba(0, 0, 0, 0.13);
    }
  }
`;

export const Image = styled.button`
  margin-left: 30px;
  margin-right: 10px;
  border-radius: 50%;
  border: none;
  height: 40px;
  width: 40px;
  img {
    object-fit: cover;
    width: 100%;
    height: 100%;
    border-radius: 50%;
  }
`;

export const ButtonMenu = styled.button`
  display: flex;
  text-align: center;
  justify-content: center;
  border-radius: 50%;
  padding: 12px;
  margin-right: 16px;
  background-color: transparent;
  color: white;
  margin-left: 12px;
  border: none;
  font-size: 1.5rem;
`;

export const NavMenu = styled.ul`
  display: flex;
  width: 100vw;
  height: 100vh;
  bottom: 0;
  position: absolute;
  display: ${({ click }) => (click ? 'flex' : 'none')};
  left: ${({ click }) => (click ? 0 : '-100vw')};
  opacity: 1;
  transition: all 0.5s ease;
  background: rgba(0, 0, 0, 0.5);
  justify-content: flex-start;
`;

export const NavItem = styled.div`
  position: absolute;
  left: 0;
  width: 250px;
  height: 100vh;
  background-color: #ffffff;
`;

export const DivItem = styled.div`
  width: 234px;
  height: 40px;

  color: black;
  position: absolute;
  top: 20px;
  display: flex;
  align-items: center;
  padding: 8px;
  box-shadow: 0 1px 1px rgba(0, 0, 0, 0.19), 0 1px 5px rgba(0, 0, 0, 0.19);
  font-size: 16px;
  font-weight: normal;
`;

export const SpanItem = styled.span`
  padding-left: 8px;
`;
