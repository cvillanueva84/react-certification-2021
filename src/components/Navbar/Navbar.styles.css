/* .sidebar.active {
  left: 0;
  transition: 350ms;
} */

/* .nav-menu.active {
  left: 0;
  transition: 350ms;
} */

.nav-menu-items {
  width: 100%;
  padding: 0;
  list-style-type: none;
}

.nav-text {
  margin: 0;
  padding: 0;
  height: 50px;

  display: flex;
  justify-content: center;
}

.nav-text a {
  text-decoration: none;
  color: rgb(73, 73, 73);
  font-size: 16px;
  width: 100%;
  height: 100%;
  padding: 20px;
  display: flex;
  align-items: center;
}

.nav-text a:hover {
  background-color: #e0e0e0;
}
