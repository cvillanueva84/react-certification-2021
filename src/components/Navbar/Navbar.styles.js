import styled from 'styled-components';
import { BiSearch } from 'react-icons/bi';
import { Container } from '../../Global.styles';

export const Nav = styled.nav`
  font-size: 18px;
  position: sticky;
  top: 0;
  z-index: 999;
  height: 80px;
  background-color: #1c5476;
  /* box-shadow: 0px 5px 20px rgba(0, 0, 0, 0.5); */
  box-shadow: 0px 2px 10px rgba(0, 0, 0, 0.15);
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const NavbarContainer = styled(Container)`
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 80px;
  ${Container};
`;

export const NavIcon = styled(BiSearch)`
  margin-right: 0.8rem;
  transition: all 0.5s ease;
`;

export const Menu = styled.ul`
  display: flex;
  align-items: center;
  text-align: center;
  @media only screen and (max-width: 1000px) {
    display: flex;
    flex-direction: column;
    width: 100%;
    height: 100vh;
    position: absolute;
    top: 80px;
    left: ${({ click }) => (click ? '0' : '-100%')};
    background-color: rgba(0, 0, 0, 0.9);
    transition: all 0.5s ease;
  }
`;

export const MenuItem = styled.li`
  list-style: none;
  height: 80px;
  @media only screen and (max-width: 1000px) {
    width: 100%;
    &:hover {
      border: none;
    }
  }
`;

export const MenuItemRight = styled.li`
  list-style: none;
  margin-top: 10px;
  height: 80px;
  float: right;
`;

export const MenuLink = styled.div`
  text-decoration: none;
  font-weight: bold;
  font-size: 2rem;
  color: #fff;
  display: inline;
  justify-content: space-between;
  align-items: center;
  height: 100%;
`;
