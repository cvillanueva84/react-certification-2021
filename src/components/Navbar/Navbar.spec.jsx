import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import Navbar from './Navbar.component';
import { SearchContext } from '../../providers/Search';
import { useDebounce } from '../../utils/hooks/useDebounce';

const RenderNavWithSearchProvider = (snap) => {
  const query = 'wizeline';
  const setQuery = jest.fn();
  const isSnap = snap || false;

  const component = (
    <SearchContext.Provider value={{ query, setQuery }}>
      <Navbar />
    </SearchContext.Provider>
  );

  if (isSnap) {
    return renderer.create(component);
  }

  return render(component);
};

describe('Navbar Component', () => {
  beforeEach(() => {
    RenderNavWithSearchProvider();
  });

  it('should match snapshot', () => {
    // const snap = renderer.create(<Navbar />);
    const snap = RenderNavWithSearchProvider(true);

    expect(snap.toJSON()).toMatchSnapshot();
  });

  it('Should render a menu bar', () => {
    const menuBars = screen.getByRole('button', {
      name: 'open drawer',
    });

    expect(menuBars).toBeInTheDocument();
  });

  it('should render a search input', () => {
    const searchBox = screen.getByRole('textbox', {
      name: 'search',
    });

    expect(searchBox).toBeInTheDocument();
  });

  it('should render a checkbox for theme switch', () => {
    const component = screen.getByRole('checkbox', {
      name: 'Dark mode',
    });

    expect(component).toBeInTheDocument();
  });


  it('Input should change should not be empty', () => {
    const searchBox = screen.getByRole('textbox');
    expect(searchBox.value).not.toBe('');
  });
  

  it('Input should change query value', () => {
    const searchBox = screen.getByRole('textbox');
    const preValue = searchBox.value;
    fireEvent.change(searchBox,{target: { value: `${preValue} Rocks!`}});

    // const spyChangeValue = jest.spyOn(Navbar,"changeLocalQuery");
    // jest.spyOn(Navbar, 'changeLocalQuery').mockImplementation((e) => {
    //   return e;
    // })
    // const spyChangeValue = jest.spyOn(navBar);
    
    expect(searchBox.value).toBe('wizeline Rocks!');
    // expect(spyChangeValue).toHaveBeenCalled();
  });

});
