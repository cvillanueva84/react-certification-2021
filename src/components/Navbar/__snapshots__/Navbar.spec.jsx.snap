// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component should match snapshot 1`] = `
<nav
  className="sc-bdnxRM jhGmdS"
>
  <div
    className="sc-gtsrHT bfKaBC"
  >
    <button
      aria-label="open drawer"
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </span>
    </button>
    <div
      className="sc-fujyAs ctKQWQ"
      onClick={[Function]}
    >
      <nav
        className="sc-pNWdM cjYudH"
        onClick={[Function]}
      >
        <ul
          className="sc-jrsJWt gyfqor"
        >
          <li
            className="sc-kEqXSa fsCpBf"
          >
            <a
              className="sc-iqAclL gnEzMY"
              href="/"
            >
              Home
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      className="sc-eCApnc ktiLQa"
    >
      <span
        className="sc-gKAaRy iXzfoV"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </span>
      <input
        aria-label="search"
        className="sc-jSFjdj fsuZxZ"
        onChange={[Function]}
        placeholder="Search..."
        value="wizeline"
      />
    </div>
  </div>
  <div
    className="sc-dlnjwi gNnYvo"
  >
    <label
      className="MuiFormControlLabel-root"
    >
      <span
        className="MuiSwitch-root"
      >
        <span
          aria-disabled={false}
          className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiSwitch-switchBase MuiSwitch-colorPrimary"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label"
          >
            <input
              checked={false}
              className="PrivateSwitchBase-input-4 MuiSwitch-input"
              name="toggleTheme"
              onChange={[Function]}
              type="checkbox"
            />
            <span
              className="MuiSwitch-thumb"
            />
          </span>
        </span>
        <span
          className="MuiSwitch-track"
        />
      </span>
      <span
        className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        Dark mode
      </span>
    </label>
    <button
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <span
          className="sc-iCoGMd WqZlr"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            style={
              Object {
                "fontSize": 27,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </span>
      </span>
    </button>
  </div>
</nav>
`;
